src/puya/context.py

```diff
@@ -16,13 +17,17 @@ class SourceMeta:
 _EmptyMeta = SourceMeta(None, None)
 
 
-@attrs.define
+@attrs.define(slots=False)
```

Comment:
we should update `attrs` so this isn't required https://github.com/python-attrs/attrs/pull/1200